packages/strapi-design-system/src/Json/JsonInput.js

```diff
@@ -0,0 +1,156 @@
+
+export const JsonInput = ({
+  id,
+  label,
+  value,
+  error,
+  hint,
+  required,
+  theme,
+  onChange,
+  disabled,
+  labelAction,
+  ...boxProps
+}) => {
+  const editor = useRef();
+  const editorState = useRef(null);
+  const editorView = useRef(null);
+
+  const getContentAtLine = (line) => {
+    return editorState.current?.doc?.line(line);
+  };
+
+  /**
+   * @description
+   * Determines the line to highlight when validateJson finds an error via jsonParseLinter()
+   * @param {number} lineNumber Code editor line number
+   */
+  const highglightErrorAtLine = (lineNumber) => {
+    const { text, to: lineEnd } = getContentAtLine(lineNumber);
+    const lineStart = lineEnd - text.trimStart().length;
+
+    if (lineEnd > lineStart) {
+      editorView.current?.dispatch({
+        effects: addMarks.of([lineHighlightMark.range(lineStart, lineEnd)]),
+      });
+    }
+  };
+
+  const clearErrorHighlight = () => {
+    const docEnd = editorState.current?.doc?.length || 0;
+    editorView.current?.dispatch({
+      effects: filterMarks.of((from, to) => to <= 0 || from >= docEnd),
+    });
+  };
+
+  /**
+   * @description
+   * Checks code editor for valid json input and then handles any errors
+   * @param {object} viewUpdate
+   * @property {object} viewUpdate.view Code editor view https://codemirror.net/docs/ref/#view.EditorView
+   * @property {object} viewUpdate.state Code editor state https://codemirror.net/docs/ref/#state.EditorState
+   * @returns {boolean} true if valid json, false if invalid json
+   */
+  const validateJson = (viewUpdate) => {
+    const { view, state } = viewUpdate;
+    editorView.current = view;
+    editorState.current = state;
+
+    clearErrorHighlight();
+
+    // Function calls json.parse and returns error message + position
+    const lintJson = jsonParseLinter();
+    const lintErrors = lintJson(view);
+
+    if (lintErrors.length) {
+      highglightErrorAtLine(state.doc.lineAt(lintErrors[0].from).number);
+    }
+  };
+
+  const handleChange = (currentValue, viewUpdate) => {
+    validateJson(viewUpdate);
+
+    // Call the parent's onChange handler
+    onChange(currentValue);
+  };
+
+  const onCreateEditor = (view, state) => {
+    editorView.current = view;
+    editorState.current = state;
+    validateJson({ view, state });
+  };
+
+  const { setContainer } = useCodeMirror({
+    theme,
+    value,
+    onCreateEditor,
+    onChange: handleChange,
+    editable: !disabled,
+    container: editor.current,
+    extensions: [json(), markField],
+    basicSetup: {
+      lineNumbers: true,
+      bracketMatching: true,
+      closeBrackets: true,
+      indentOnInput: true,
+      syntaxHighlighting: true,
+      highlightSelectionMatches: true,
+      tabSize: 2,
+      defaultCharacterWidth: 5,
+    },
+  });
+
+  useEffect(() => {
+    const currentEditor = editor.current;
+
+    if (currentEditor) {
+      setContainer(editor.current);
+    }
+  }, [setContainer]);
```

Comment:
Should `editor.current` be a dependency here, so the effect runs once it changes?

Reply:
I had it but the linter was telling me it shouldn't be there 🤔 

![Screenshot 2023-01-09 at 11 02 13](https://user-images.githubusercontent.com/26598053/211283051-230b1dbd-e893-4827-81d3-4b3b7863b37a.png)

Reply:
Ok, so better believe eslint than me!